api/envoy/config/cluster/v3/cluster.proto

```diff
@@ -159,6 +159,17 @@ message Cluster {
     USE_DOWNSTREAM_PROTOCOL = 1;
   }
 
+  // Refer to :ref:`on_demand_dns_mode <envoy_v3_api_field_config.cluster.v3.Cluster.on_demand_dns_mode>`
+  // for an explanation of each type.
+  enum OnDemandDNSMode {
```

Comment:
Do you anticipate adding additional modes? If not, this could probably just be a boolean.

Reply:
Yes, I hope to eventually add a mode where the DNS resolution can back off if there are fewer connections for the cluster (option 1 in the linked issue)

Reply:
What would the full configuration look like with that mode? I presume it would also need a threshold.

Reply:
Presumably, that mode would utilize the refresh rate/DNS record TTL, but if users wanted, we could allow them to set a threshold.

Reply:
We might want this to be a `oneof` between messages to give future growth possibility in that case.

Reply:
To clarify: you mean `oneof` between two messages instead of of an enum with options?

Reply:
Yes, although I now remember that we prefer to not use `oneof` for reasons outlines at https://github.com/envoyproxy/envoy/blob/main/api/STYLE.md#:~:text=Prefer%20multiple%20fields%20with%20defined%20precedence%20over%20boolean%20overloads%20of%20fields%20or%20oneof.%20For%20example%2C%20prefer%3A.

So, it would just be separate messages, their presence will indicate which one to use (with precedence). No need for a disable one.

Reply:
So since we only have 1 option at the moment (excluding disabled), would we just create one (message,field) pair now and add another one later? I'm not sure how that's better than using the enum and then creating a new field if/when we decide to add another mode

Reply:
Yes